python/scripts/rfi_receiver/rfi_receiver.py

```diff
@@ -810,109 +836,69 @@ def rfi_zeroing():
+        )
+        logger.info(
+            "RFI Solar Transit Toggle: Time of nearest transit: {}".format(
+                datetime.datetime.fromtimestamp(time_to_nearest_transit + time_now)
             )
+        )
 
-            time.sleep(sleep_time_s)
+        new_zeroing_state = True
 
+        # Check if we are within the current transit window and wait until the end of it
+        if abs(time_to_nearest_transit) < half_window_s:
+            new_zeroing_state = False
+            downtime_s = half_window_s + time_to_nearest_transit
             logger.info(
-                "RFI Solar Transit Toggle: Waking up to disable RFI zeroing during solar transit"
-            )
-
-        # Create payload
-        payload = {"rfi_zeroing": False}
-
-        # Turn RFI zeroing off
-        rfi_zeroing_on = True
-        try:
-            r = requests.post(url, data=json.dumps(payload), headers=headers)
-            if not r.ok:
-                logger.info(
-                    "RFI Solar Transit Toggle: Failed to turn RFI zeroing off. Something went wrong in the request."
+                "RFI Solar Transit Toggle: Within solar transit window, disabling zeroing and sleeping for {} seconds until end of window.".format(
+                    downtime_s
                 )
-            else:
-                rfi_zeroing_on = False
-                logger.info(
-                    "RFI Solar Transit Toggle: Successfully turned RFI zeroing off."
-                )
-        except:
-            logger.info(
-                "RFI Solar Transit Toggle: Failure to contact Comet, is it running?"
             )
-
-        # If we successfully turned RFI zeroing off
-        if not rfi_zeroing_on:
-
+        # Otherwise, we wait until the start of the next transit window
+        else:
+            new_zeroing_state = True
+            downtime_s = time_to_next_transit - half_window_s
             logger.info(
-                "RFI Solar Transit Toggle: Sleeping %s seconds for duration of solar transit."
-                % (downtime_s)
+                "RFI Solar Transit Toggle: Outside solar transit window, enabling zeroing and sleeping for {} seconds until next window.".format(
+                    downtime_s
+                )
             )
 
-            # Wait until sun has passed
-            time.sleep(downtime_s)
+        # Set new RFI zeroing state
+        success = set_rfi_zeroing(new_zeroing_state)
+
+        # If we failed to set new RFI zeroing state sleep for a few seconds
+        if not success:
 
             logger.info(
-                "RFI Solar Transit Toggle: Solar transit ended. Waking up to turn RFI zeroing back on."
+                "RFI Solar Transit Toggle: Failed to set new RFI zeroing state. Will wait for a few seconds and try again."
             )
 
-            # Payload
-            payload = {"rfi_zeroing": True}
+            time.sleep(5)
```

Comment:
```suggestion
            time.sleep(5)
            continue
```
At the moment this doesn't do anything, since you'll get into the next sleep and never set the state.  I think either rerunning the loop, or exiting (and letting systemd restart it) should do the trick though.